src/suspendplugin.c

```diff
@@ -194,23 +217,13 @@ suspend_set_wowlan(
 
     nla_nest_end(msg, wowlan_triggers);
 
-    err = 1;
-    nl_cb_err(cb, NL_CB_CUSTOM, handle_nl_command_error, &err);
-    nl_cb_set(cb, NL_CB_VALID, NL_CB_CUSTOM, handle_nl_command_valid, &err);
-    nl_cb_set(cb, NL_CB_FINISH, NL_CB_CUSTOM, handle_nl_command_finished, &err);
-    nl_cb_set(cb, NL_CB_ACK, NL_CB_CUSTOM, handle_nl_command_ack, &err);
-
-    if (nl_send_auto_complete(nl_socket, msg) < 0) {
+    if (nl_send_auto(nl_socket, msg) < 0) {
         connman_error("Failed to send wowlan command.\n");
     }
 
-    while (err > 0) {
-        DBG("Waiting until wowlan command has been finalized.\n");
-        nl_recvmsgs(nl_socket, cb);
-    }
+    suspend_plugin_netlink_handler();
```

Comment:
Assuming nl_send_auto() sends request and suspend_plugin_netlink_handler() waits for it to get processed, does it make sense to wait for result if sending the request failed?

Reply:
no :)